packages/web-components/fast-foundation/src/menu-item/menu-item.template.ts

```diff
@@ -30,25 +30,21 @@ export function menuItemTemplate<T extends FASTMenuItem>(
             ${when(
```

Comment:
This is a place where multiple `when` directives could be replaced with a simple template cache lookup based on role.

Reply:
As discussed offline, I think we'll move the role specific stuff to be a single slot after this PR. If different SVG's are needed by default that can be done/customized by passing conditionals in the implementation.

Reply:
Having a single slot doesn't seem worth the tradeoff here. They are commonly going to be different visually, and needing to slot that in each time would be quite a hassle. I believe there was an example of the template swap somewhere, so I'll look for that and we can start there.

Reply:
> They are commonly going to be different visually, and needing to slot that in each time would be quite a hassle.

I don't actually think this is the case. The delta here is behavioral and not visual. While some implementations *may* want to differentiate based on behavior, I don't agree with the idea that having unique indicators for each here is actually common enough to warrant two distinct slots. I'll add that I'm not sure what the tradeoff is in terms of implementation here, we're simply *improving* the default experience by flattening into a single instance _by default_. That offers improved performance for those who need a single visual indicator for illustrating something is _checked_. Those who need different visuals can still accomplish that with relative ease. 

In terms of adding more to this PR, can we please limit that? It's going to prolong this being outstanding as we continue to add more and more. I think it's probably ideal to take the further refactoring into a different PR.